### PR TITLE
[many] Upgrade example apps to AGP 8.5.2 (if they were below 8.1.0)

### PR DESCRIPTION
**Pull request by gmackall**
For https://github.com/flutter/flutter/pull/153795 - specifically the bump to `compileSdk 35` requires AGP 8.1.0+ (this isn't documented anywhere, but folks on https://issuetracker.google.com/issues/342522139 are also experiencing it).

Upgrades to AGP 8.5.2 and Gradle 8.7, for example apps that were using AGP less than 8.1.0.

Unblocks framework->packages roller

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: .ci/legacy_project/all_packages/android/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.0'
+        classpath 'com.android.tools.build:gradle:8.1.0'
```
Comment (gmackall) [inline .ci/legacy_project/all_packages/android/build.gradle:8]: @stuartmorgan Do you have any opposition to upgrading the legacy project AGP version here?

The reason the upgrade is required is that the AGP version is shared by the legacy and master versions of the test, and the master version requires AGP 8.1.0.

Comment (stuartmorgan) [reply .ci/legacy_project/all_packages/android/build.gradle:8]: That means we won't notice if we break people on `stable`, which is not ideal. How hard would it be to have the repo tooling adjust this after copying, based on the branch?

Comment (gmackall) [reply .ci/legacy_project/all_packages/android/build.gradle:8]: I can do a simple find and replace somewhere in `create_all_packages_app_command.dart`, perhaps somewhere in https://github.com/flutter/packages/blob/bcb09dbc121e5f886a37cf2808c0d95112276fcd/script/tool/lib/src/create_all_packages_app_command.dart#L130

Is that sort of what you had in mind?

Comment (gmackall) [reply .ci/legacy_project/all_packages/android/build.gradle:8]: Or plumbed through all the way to be part of the args for the command? Something like `--agp-override=7.0.1`?

Comment (gmackall) [reply .ci/legacy_project/all_packages/android/build.gradle:8]: Actually - the legacy project doesn't use the `FlutterExtension` for the compileSdk version. So I can just remove this upgrade

**Review by Piinks — APPROVED** [review]
LGTM for two_dimensional_scrollables

**Review by matanlurey — APPROVED** [review]

**Review by reidbaker — APPROVED** [review]

**Comment by matanlurey** [comment]
test-exempt: The example apps are the tests.

**Comment by gmackall** [comment]
Overriding the repo checks: The only auto-exempted change is a deletion of a gradle wrapper that didn't need to exist.

**Pull request merged**